cpp/include/cudf/io/types.hpp

```diff
@@ -191,19 +191,35 @@ struct source_info {
  * @brief Destination information for write interfaces
  */
 struct sink_info {
-  io_type type = io_type::VOID;
-  std::string filepath;
-  std::vector<char>* buffer      = nullptr;
-  cudf::io::data_sink* user_sink = nullptr;
+  io_type type          = io_type::VOID;
+  size_t num_void_sinks = 1;
+  std::vector<std::string> filepaths;
+  std::vector<std::vector<char>*> buffers;  // TODO: perhaps we can repurpose host_buffer. ask VM
```

Comment:
Unfortunately we cannot, because writer allocates the memory in the output buffer(s). `host_buffer` is just a view into existing memory. Would love to see a more generic type here.